src/colmap/scene/frame.h

```diff
@@ -114,10 +96,13 @@ class Frame {
   // case, where rig modeling is no longer needed.
   std::optional<Rigid3d> frame_from_world_;
 
-  // [Optional] Rig calibration
-  std::shared_ptr<class Rig> rig_ = nullptr;
+  // Rig calibration.
+  rig_t rig_id_ = kInvalidRigId;
```

Comment:
Does the frame need to store the rig id given that it is already stored in the rig? If so, consider documenting why. I understand that it is convenient to only populate the rig id before calling `Reconstruction::AddFrame`, but is this really necessary?

Reply:
This is necessary for the same reason we need to store the camera_id separately from the camera_ptr in the image: When loading the data from the database (or from disk), we do not yet have the object available to assign the right object pointers.